endpoints/internal-check-file-acl.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Automattic\VIP\Files\Acl;
+
+require_once __DIR__ . '/../files/acl.php';
+
+$file_request_uri = $_SERVER['HTTP_X_ORIGINAL_URI'] ?? null;
+
+$validated_file_path = pre_wp_validate_request( $file_request_uri );
```

Comment:
This part is kind of unobvious:
1. We expect the file path to be a string.
2. But then the validation can turn it into an array if the file is invalid.
3. But if it's valid we also normalize and sanitize the string in the validation function.

This introduces a bunch of ambiguity, which can make it hard to follow the logic.

I propose splitting `pre_wp_validate_request` in two functions:
1.  `pre_wp_validate_request` would always return a tuple with code and message, 200 means valid)
2. `sanitize_file_path` (or `normalize_file_path`) would then do `Strip off subdirectory` part.

This way we can get a consistent return signature and make it easier to follow.

Reply:
I originally had it returning a `WP_Error` object but then realized that WP is not loaded 🤦 

I was a bit unsure about doing both validation and sanitization in the same function but ended up going that way for simplicity by minimizing the number of functions + branches (they're not written for re-use, just for encapsulation and testability) and to avoid duplicating work (we'd have to call `parse_url` twice, once in the validation function and once in the sanitization one).

I could rename the function (`pre_wp_validate_and_sanitize_original_uri`?) to add more clarity if that would help?

Reply:
I ended up separating them and moving both to a separate namespace so their use and limitations are more obvious.